### PR TITLE
Adding pytest automation

### PR DESCRIPTION
**Pull request by Chris-Schnaufer**
## Description

Added pytest tests using JSON and TXT files as input
Changes to plotclip.py to support exception testing
Added testing input files
Added matrix to GitHub Workflow to run pylint and pytest
Renamed Workflow file to match testing outside of pylint

## Types of changes

Addition of tests

## Checklist:

- [x] I have read the documentation on [CONTRIBUTING](https://github.com/AgPipeline/Organization-info) 
- [ ] My updates require a change to the documentation 
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## Time frame for completing Pull Request review (from creation date):
- [ ] As soon as possible
- [x] In the next two weeks
- [ ] In the next month

## Focus of Pull Request:

Reviewers should focus on the following aspects of this pull request. All comments are welcome regardless of which items are checked off.
- [x] Inform others of new features and changes
- [ ] There are no errors or inconsistencies
- [ ] Everything appears clear and easy to understand
- [ ] Suggest improvements
- [ ] Identify areas where functionality and/or documentation is missing

## Hints to reviewers:

Not complete code coverage
A sample of running the existing tests, under automation, can be found at https://github.com/AgPipeline/transformer-soilmask/actions/runs/222428117
@julianpistorius Review changes
@dlebauer Review README, can download artifact report from above link, and whatever else you want
@jvanderleeuw Review changes

## Optional: issue links this Pull Request resolves

Part of https://github.com/AgPipeline/issues-and-projects/issues/235, https://github.com/AgPipeline/issues-and-projects/issues/238, https://github.com/AgPipeline/issues-and-projects/issues/247

### CODE REVIEW

File: soilmask.py [hunk 1]
```diff
@@ -42,7 +43,8 @@ def prepare_metadata_for_geotiff(transformer_info: dict = None) -> dict:
             extra_metadata["transformer_version"] = str(transformer_info.get("version", ""))
             extra_metadata["transformer_author"] = str(transformer_info.get("author", ""))
             extra_metadata["transformer_description"] = str(transformer_info.get("description", ""))
-            if "repository" in transformer_info and "repUrl" in transformer_info["repository"]:
+            if "repository" in transformer_info and transformer_info["repository"] and \
```
Comment (julianpistorius) [inline soilmask.py:46]:
```suggestion
            extra_metadata["transformer_repo"] = str(transformer_info.get("repository", {}).get("repUrl", ""))
```

File: soilmask.py [hunk 2]
```diff
@@ -420,3 +422,4 @@ def perform_process(self, environment: Environment, check_md: dict, transformer_
 if __name__ == "__main__":
     CONFIGURATION = ConfigurationSoilmask()
     entrypoint.entrypoint(CONFIGURATION, SoilMask())
+    sys.exit(0)
```
Comment (julianpistorius) [inline soilmask.py:425]: This is redundant and can be removed. A successful Python script will exit with code 0.

File: soilmask.py (hunk 1 above)
Comment (Chris-Schnaufer) [reply soilmask.py:46]: @julianpistorius This doesn't work for: `transformer_info = {"repository":None}`
Screen Shot 2020-08-28 at 12 30 01 PM

File: soilmask.py (hunk 2 above)
Comment (Chris-Schnaufer) [reply soilmask.py]: Changed

File: soilmask.py (hunk 1 above)
Comment (julianpistorius) [reply soilmask.py:45]: Oh, yeah, sorry. My suggestion will only work if the key `"repository"` doesn't exist at all in the `transformer_info` dictionary, but will fail if it exists and is set explicitly to None.

This should work, but now it's getting kinda ugly and less readable than your original:

```python
str((transformer_info.get("repository", {}) or {}).get("repUrl", ""))
```

**Review by julianpistorius — APPROVED** [review]

**Pull request merged**